app/controllers/concerns/signature_verification.rb

```diff
@@ -98,6 +98,15 @@ def signed_request_account
   rescue SignatureVerificationError => e
     @signature_verification_failure_reason = e.message
     @signed_request_account = nil
+  rescue HTTP::Error, OpenSSL::SSL::SSLError => e
+    @signature_verification_failure_reason = "Failed to fetch remote data: #{e.message}"
+    @signed_request_account = nil
+  rescue Mastodon::UnexptectedResponseError
+    @signature_verification_failure_reason = "Failed to fetch remote data (got unexpected reply from server)"
+    @signed_request_account = nil
+  rescue Stoplight::Error::RedLight
+    @signature_verification_failure_reason = "Fetching attempt skipped because of recent connection failure"
+    @signed_request_account = nil
```

Comment:
Is there a way to do this less verbosely? This is a lot of repetition now. How about capturing all errors in one rescue and then having a method that returns the human-readable message by using `case` on the `e.class`?

Reply:
I refactored it a bit with a new helper method to set the error message, resulting in shorter code. I did not change the catching logic as I don't think using a single catch clause + a case would be significantly more concise, and would be more of a mess regarding subclasses etc.